src/Controller/Component/FlashComponent.php

```diff
@@ -51,18 +43,6 @@ class FlashComponent extends Component
         'duplicate' => true
     ];
 
-    /**
-     * Constructor
-     *
-     * @param \Cake\Controller\ComponentRegistry $registry A ComponentRegistry for this component
-     * @param array $config Array of config.
-     */
-    public function __construct(ComponentRegistry $registry, array $config = [])
-    {
-        parent::__construct($registry, $config);
-        $this->_session = $registry->getController()->getRequest()->getSession();
```

Comment:
We should leave this behind and just stop using it internally. App extensions might still be using it.